src/vue-testing-library.js

```diff
@@ -91,10 +91,16 @@ function cleanupAtWrapper(wrapper) {
   mountedWrappers.delete(wrapper)
 }
 
-Object.keys(fireEvent).forEach(fn => {
-  fireEvent[`_${fn}`] = fireEvent[fn]
-  fireEvent[fn] = async (...params) => {
-    fireEvent[`_${fn}`](...params)
+// Vue Testing Library's version of fireEvent will call DOM Testing Library's
+// version of fireEvent plus wait for one tick of the event loop so that...
```

Comment:
"so that..."

couldn't think of a good wording to explain why waiting for a tick is needed. Any ideas?

Reply:
I'd love to know, too, whether if making fireEvent return a promise is totally necessary due to Vue internals. There might be some alternative that keeps VTL public API as similar as possible to RTL's and DTL's.

Reply:
>Any ideas?

How about "to allow vue to asynchronously handle the event"?

Maybe it would make sense to link to the docs here: https://vuejs.org/v2/guide/reactivity.html#Async-Update-Queue

Good question on if it is necessary for fireevent to be async. Both React and Preact expose ways of forcing them to flush their update queues synchronously. Maybe it is worth looking into [`vm.$forceUpdate`](https://vuejs.org/v2/api/#vm-forceUpdate) to see if it does what we want

Reply:
> How about "to allow vue to asynchronously handle the event"?

Sounds good! Added.

I'm gonna give `vm.$forceUpdate` (or even `nextTick`? who knows) a go. I'd really love to stick to synchronous events.

Reply:
Might also be worth looking into https://github.com/vuejs/vue-test-utils/issues/1137